### PR TITLE
Cleanup jobs modules

### PR DESCRIPTION
**Pull request by FlorianSW**
The following changes was made to scripts in the jobs directory of this project:

* Changed big if/else control structure for error management to if + early
  return This makes the code a bit easier to read as the indention level is
  kept as low as possible
* removed commented code
* removed unused imports/requires and unused declared variables
* for a consistent use: changed all " to '
* merged multiple var definition blocks
* moved variable declarations to the top of modules/functions
* Changed usage of type-converting comparsion (equality, ==) to strict comparsion (strict
  equality, ===) where usage of type-converting comparsion doesn't seem to be explicitly
  needed. The same applies for non-equality comparsions (!= -> !==)

In this commit, the every5minstat.js script is rewritten to fulfill the things above and
make the code a lot smaller.

### CODE REVIEW

**Comment by marziman** [comment]
@FlorianSW,
I will review this later..

**Comment by marziman** [comment]
@FlorianSW I tried to check your PR but I have merge conflicts.
Are you going to clean them up?
Thanks mate!
BR Mehmet

**Comment by FlorianSW** [comment]
Hi @marziman. It seems, that the branch is mergeable with the openhab/openhab-cloud@master, so probably you made some changes to your copy locally?

**Comment by digitaldan** [comment]
@FlorianSW the branch works fine for me.

**Comment by marziman** [comment]
@FlorianSW got it!
I had some changes and I am now back on track.
Testing and checking out the cool stuff :-)

File: jobs/checkopenhabsoffline.js [hunk 1]
```diff
@@ -1,64 +1,67 @@
...
+
+        if (!openhabs) {
+            logger.info('openHAB-cloud: No offline openHABs found');
+        }
+        logger.info('openHAB-cloud: Found ' + openhabs.length + ' openhabs');
+        for (var i in openhabs) {
+            var openhab = openhabs[i];
+
+            if (openhab.last_email_notification && openhab.last_email_notification > date3DaysAgo) {
+                continue;
```
Comment (marziman) [inline jobs/checkopenhabsoffline.js]: @FlorianSW wow great! The code is much more readable and cleaner now!
Great job! I will deep more into this after merging your other PR.

File: jobs/every5minstat.js [hunk 2]
```diff
@@ -1,55 +1,90 @@
...
+            stats['invitationUsedCount'],
+            'invitationUnusedCount',
+            stats['invitationUnusedCount'],
+            'userDeviceCount',
+            stats['userDeviceCount'],
+            'last5MinStatTimestamp',
+            new Date()
+        ],
+        function (err, result) {
+            logger.info('openHAB-cloud: every5min statistics collection job finished');
```
Comment (marziman) [inline jobs/every5minstat.js]: @FlorianSW Same here. Thanks for your effort.
Will test this asap on my instance.

File: jobs/every5minstat.js [hunk 3]
```diff
@@ -1,55 +1,90 @@
...
-    });
+    );
+}
+
+module.exports = new cronJob('00 */5 * * * *', function () {
+    logger.info('openHAB-cloud: every5min statistics collection job started');
+
+    // number of invocations of the finishSent() function to know, when all requests are finished to track multiple
+    // async actions
+    expectedFinishCount = 6;
```
Comment (digitaldan) [inline jobs/every5minstat.js]: So i posted this 2 days ago, but did not click the "start review" button, that's twice that new feature from github has bit me.

Comment (FlorianSW) [inline jobs/every5minstat.js]: Haha, I thought mongoose can't return a promise (that's why I implemented this tracking with which I'm not very happy myself at all). Now I checked this a bit more and it seems, exec returns a promise, so I'll take a look into this, thanks for the note!

Comment (FlorianSW) [inline jobs/every5minstat.js]: That is a way better solution, thanks again for pointing me to it! :)

File: jobs/every5minstat.js [hunk 4]
```diff
@@ -1,55 +1,90 @@
...
+    logger.info('openHAB-cloud: every5min statistics collection job started');
+
+    promises.push(Openhab.count({}, countCallback.bind('openhabCount')).exec());
+    promises.push(Openhab.count({status: 'online'}, countCallback.bind('openhabOnlineCount')).exec());
+    promises.push(User.count({}, countCallback.bind('userCount')).exec());
+    promises.push(Invitation.count({used:true}, countCallback.bind('invitationUsedCount')).exec());
+    promises.push(Invitation.count({used:false}, countCallback.bind('invitationUnusedCount')).exec());
+    promises.push(UserDevice.count({}, countCallback.bind('userDeviceCount')).exec());
+
+    Promise.all(promises).then(saveStats);
```
Comment (digitaldan) [inline jobs/every5minstat.js]: @FlorianSW LGTM,  I just need to run it on my test instance tonight or tomorrow.

Comment (FlorianSW) [inline jobs/every5minstat.js]: no hurry, thanks for your review! :)

**Comment by digitaldan** [comment]
Thanks!

**Pull request merged**